### DISCUSSION TITLE
Troubleshooting Grouped Flow Filtering Issue in nfdump: Seeking Assistance and Ideas

### DISCUSSION THREAD

**Discussion by piualves** [phaag/nfdump#450] [Q&A] [answered]
Hello,

I'm not sure if I understood the nfdump manual correctly, where it tells me that I can apply a filter based on the number of flows (provided they are grouped by the options -A, -B, or -b). I'm experiencing an issue when trying to use this type of filter, as it doesn't seem to be based on the grouped flow count.

For example, when I execute nfdump with the filter (e.g., nfdump -R exporter_1/ -B -O flows -o "extended" 'flows > 10'), I don't get any results, and the command returns "No matching flows." However, when I use the filter 'flows = 1,' it works perfectly fine, and the "Flows" column shows the correct grouping.

Has anyone encountered a similar problem with this type of filter? Any ideas?

**Comment by phaag** [comment, marked as answer]
The point is the order of the flow processing chain, how the individual steps are executed.
1. The given filter is applied to the flows, read from file.
2. The matching flows are aggregated in the flow cache.
3. The flows in the flow cache are sorted and exported.

In your case, it means you would like to apply the filter *after* aggregation. In order to make this work, you need to create an intermediate file from the exported flow cache. For that, nfdump allows you to write the aggregated flow cache in a new flow file:

`nfdump -r exporter_1/ -B -w tmpFile`
`nfdump -r tmpFile  -O flows -o extended 'flows > 10'`

By creating this intermediate file, you can filter according to your needs.

**Reply by piualves** [reply]
It worked perfectly! Thank you for the help and explanations.
